contexts/NotificationsContext.tsx

```diff
@@ -4,28 +4,35 @@ import Notifications from '@/components/Notifications';
 export type TxNotification = {
   id: string;
   hash: `0x${string}`;
-  functionName: string;
+  functionName?: string;
   txStatus: 'pending' | 'success' | 'reverted';
-  chainId: number;
+  chainId?: number;
+  description?: string;
   isCrossChain?: boolean;
 };
 
 type NotificationsContextType = {
   notifications: TxNotification[];
-  notify: Function;
-  dismiss: Function;
+  notify: (arg0: Omit<TxNotification, 'id'>) => void;
```

Comment:
if we're not specifying an ID in `notify`, then let's remove the `id` from the `TxNotification` type, and remove the `Omit` utility type


```suggestion
  notify: (arg0: TxNotification) => void;
```

Reply:
There is one case where we want `id` specified and that is in the input argument of `setNotification`. There are a couple options. We could move the Omit type into a separate type, we could join an id type to a TxNotification type without the id field. Open to changing it, but I think we will need two types.